src/envoy/tcp/mixer/filter.cc

```diff
@@ -140,7 +138,7 @@ Network::FilterStatus Filter::onNewConnection() {
 void Filter::completeCheck(const CheckResponseInfo &info) {
   const auto &status = info.status();
   ENVOY_LOG(debug, "Called tcp filter completeCheck: {}", status.ToString());
-  cancel_check_ = nullptr;
+  handler_->ResetCancel();
```

Comment:
Is the `completeCheck` always invoked after `callCheck`(so that handler_ is not nullptr)?

Reply:
I think callCheck is always invoked before completeCheck.  I'm pretty unfamiliar with this code, but other code in the same function calls handler_ without checking for null.